crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -416,7 +435,10 @@ fn convert_onchain_order_placement(
         buy_amount: u256_to_big_decimal(&order_data.buy_amount),
         valid_to: order_data.valid_to as i64,
         app_data: ByteArray(order_data.app_data.0),
-        fee_amount: u256_to_big_decimal(&order_data.fee_amount),
+        fee_amount: match class {
+            OrderClass::Limit => Default::default(),
+            _ => u256_to_big_decimal(&order_data.fee_amount),
+        },
```

Comment:
This and some code below makes the assumption that `order_data.fee_amount` will be the surplus fee for limit orders and regular fee for other orders. I think this is the right approach, please correct me if I'm wrong.

Reply:
I believe it's required to leave the `fee_amount` untouched to not run into [unforeseen complications](https://github.com/cowprotocol/services/pull/713#discussion_r1012956432) down the line.
AFAICS this would be okay at the moment since all the limit orders are supposed to have 0 fee but should we ever charge a flat fee on top of the dynamic fees your are computing in this PR things would get weird.

I don't know for sure but I believe the solutions coming from the solvers would fail to simulate because we try to execute an order with parameters that are different than what the user signed (unless we have another place in the code which would undo this `fee_amount` change again).

Reply:
Alright so that's getting hairy. Let's leave it like this and I'd say in the future we might want to just get the order ID and use that to fetch the order from the DB and then compare the other fields as a sanity check. This might simplify things.

Reply:
Sorry, I realize that my comment ☝️ may be a bit confusing. Concretely my suggestion for `database/onchain_order_events.rs` is:

```rust
let order = database::orders::Order {
    fee_amount: u256_to_big_decimal(&order_data.fee_amount),
    surplus_fee: BigDecimal::zero(),
    ..order
};
```

As the surplus fee is not calculated at order indexing time, and the order's fee amount we actually store needs to be what the on-chain order is emitting (otherwise, the order we encode when submitting the actual transaction will be invalid).